reducer/src/main/java/com/graphicsfuzz/reducer/reductionopportunities/Compatibility.java

```diff
@@ -91,6 +91,18 @@ public static boolean compatible(Class<? extends IReductionOpportunity> first,
       return false;
     }
 
+    if (matchesEitherDirection(first, second,
+        StmtReductionOpportunity.class::equals,
+        UnswitchifyReductionOpportunity.class::equals)) {
+      return false;
+    }
```

Comment:
This looks identical to the above. Can remove? Maybe could also add some comments to the ones where you can remember the reasons?

Reply:
Removed the duplication - I guess that must have arisen due to a previous change introducing the same compatibility check.

Unfortunately the reasons for these various checks have escaped me, and it wouldn't surprise me if some of them are no longer needed, due to having changed how the reducer works.  It would be worth digging into them in depth again at some point, removing those that do not seem to be needed and documenting why the others are needed.